conftest.py

```diff
@@ -856,3 +771,132 @@ def num_calls(self) -> int:
+
+@pytest.fixture(scope="function")
+def local_experiment(test_dir, wlmutils):
+    """Create a default experiment that uses the requested launcher"""
+    name = pathlib.Path(test_dir).stem
+    return smartsim.Experiment(name, exp_path=test_dir, launcher="local")
+
+@pytest.fixture(scope="function")
+def wlm_experiment(test_dir, wlmutils):
+    """Create a default experiment that uses the requested launcher"""
+    name = pathlib.Path(test_dir).stem
+    return smartsim.Experiment(name, exp_path=test_dir, launcher="local")
+
+def _cleanup_db(name):
+    global database_registry
+    db = database_registry[name]
+    if db and db.is_active():
+        exp = Experiment("cleanup")
+        try:
+            db = exp.reconnect_orchestrator(db.checkpoint_file)
+            exp.stop(db)
+        except:
+            pass
+
+# Reuse databases
+@pytest.fixture(scope="session")
+def local_db():
+    name = "local_db_fixture"
+    config = dict(
+        name = name,
+        launcher = "local",
+        num_nodes = 1,
+        interface = "lo",
+        hostlist = None,
+        port = _find_free_port(tuple(reversed(test_ports))),
+    )
+    yield config
+    _cleanup_db(name)
+
+@pytest.fixture(scope="session")
+def single_db(wlmutils):
+    hostlist = wlmutils.get_test_hostlist()
+    hostlist = hostlist[-1:] if hostlist is not None else None
+    name = "single_db_fixture"
+    config = dict(
+        name = name,
+        launcher = wlmutils.get_test_launcher(),
+        num_nodes = 1,
+        interface = wlmutils.get_test_interface(),
+        hostlist = hostlist,
+        port = _find_free_port(tuple(reversed(test_ports)))
+    )
+    yield config
+    _cleanup_db(name)
+
+
+@pytest.fixture(scope="session")
+def clustered_db(wlmutils):
+    hostlist = wlmutils.get_test_hostlist()
+    hostlist = hostlist[-4:-1] if hostlist is not None else None
+    name = "clustered_db_fixture"
+    config = dict(
+        name = name,
+        launcher = wlmutils.get_test_launcher(),
+        num_nodes = 3,
+        interface = wlmutils.get_test_interface(),
+        hostlist = hostlist,
+        port = _find_free_port(tuple(reversed(test_ports))),
+    )
+    yield config
+    _cleanup_db(name)
+
+
+@pytest.fixture
+def register_new_db():
+    def _register_new_db(
+        name: str,
+        launcher: str,
+        num_nodes: int,
+        interface: str,
+        hostlist: t.Optional[t.List[str]],
+        port: int,
+    ):
+        exp_path = pathlib.Path(test_output_root, name)
+        exp_path.mkdir(exist_ok=True)
+        exp = Experiment(
+            name,
+            exp_path=str(exp_path),
+            launcher=launcher,
+        )
+        orc = exp.create_database(
+            port, batch=False, interface=interface, hosts=hostlist, db_nodes=num_nodes
+        )
+        exp.generate(orc, overwrite=True)
+        exp.start(orc)
+        global database_registry
+        database_registry[name] = orc
```

Comment:
Does the `database_registry` hold the name of the experiment instead of the database?

Reply:
It's supposed to be the identifier for a specific "type" of orchestrator that a test requests. Since we can't launch a database outside of an experiment, the two are synonymous right now.

Reply:
Makes sense!